### ISSUE TITLE
Missing/swapped characters after processing with pdfsizeopt

### ISSUE THREAD

**Issue by rbrito** [pts/pdfsizeopt#34]
Dear @pts,

I was going through my collection of PDF files and I found one document that presents problems with a single character after being processed with pdfsizeopt.

The document in question is the following:

http://www.css.cornell.edu/faculty/dgr2/teach/R/RIntro_ITC.pdf

The problem occurs on page 113. It also occurs if I invoke pdfsizeopt with the option `--do-optimize-fonts=no`.

If I simply process the file with qpdf (with `qpdf RIntro_ITC.pdf out.pdf`), then the output looks fine. If I run pdfsizeopt on this output, the problem is still present.

If I preprocess the file with ghostscript (with `ps2pdf -dPDFSETTINGS=/prepress RIntro_ITC.pdf`), then I get warnings on its output like the following:

```
   **** Error: Invalid BaseEncoding name "PDFDocEncoding" ignoring BaseEncoding.
               Output may be incorrect.
   **** Error: Invalid BaseEncoding name "PDFDocEncoding" ignoring BaseEncoding.
               Output may be incorrect.
   **** Error: Invalid BaseEncoding name "PDFDocEncoding" ignoring BaseEncoding.
               Output may be incorrect.
   **** Error: Invalid BaseEncoding name "PDFDocEncoding" ignoring BaseEncoding.
               Output may be incorrect.
```

The file was created with pdfTeX (incredible how many files I have reported here that are produced by pdfTeX that seem to have problems), which one would believe to be one of the best PDF generators out there...

Since the file may be defective, I don't know the priority that you may attribute to this issue...

**Comment by pts** [comment]
Thank you for reporting this issue!

I was able to reproduce the problem in a smaller PDF file, the JPEG file shows in red where the glyphs are missing on the page:  [rip113.pdf](https://github.com/pts/pdfsizeopt/files/1262734/rip113.pdf)
 [rip113.pso.pdf](https://github.com/pts/pdfsizeopt/files/1262736/rip113.pso.pdf)

![rip113 psobad](https://user-images.githubusercontent.com/379746/29860281-e97d8032-8d65-11e7-89a4-e6fb71be2756.jpg)

When I zoom in, the glyph seems to be replaced with another glyph:

![rip113 psozoom](https://user-images.githubusercontent.com/379746/29860515-9796b486-8d66-11e7-92f8-c47118359f7a.jpg)

With `pdfsizeopt --use-pngout=no --do-optimize-images=no --do-optimize-fonts=no rip113.pdf'`, the glyph is fine for me in rip113.pso.pdf.

I'm investigating what's going on.

**Comment by pts** [comment]
Fixed in 88987a4dc0d152e1c599c6dc01eea993be0fd31e.

This is an image processing bug (rather than a font processing bug) in pdfsizeopt, and thus it can be avoided with `--do-optimize-images=no`.

The glyph is embedded as a tiny image to the document. The bug is that pdfsizeopt removes `/Decode [1 0]` from the `/Subtype /Image` obj.

**Comment by rbrito** [comment]
Thanks a lot, Péter!
